enumerable.rb

```diff
@@ -1,72 +1,111 @@
-module Enumerable
-    def my_each(arr)
-result=[]
-for i in 0..length
-    result.push(yield(arr[i]))
-end
-result
-    end
- 
-    def my_each_with_index(arr)
-        result = []
-        for i in 0..length
-            result.push(yield(arr[i],i))
-        end
-        result
-    end
-
-    def my_select
-        result=[]
-        my_each do |x|
-            output.push(x) if yield x
-        end
-        result
-    end
-
-    def my_all
-        result = true
-        my_each do |x|
-            result = false unless yield x
-        end
-        result
-    end
-
-    def my_any?
-        result = false
-        my_each do |x|
-            result = false if yield x
-        end 
-        result
-    end
-
-    def my_none?
-        result = true
-        my_each do |x|
-            result = false if yield x
-        end
-        result
-    end
-
-    def my_count
-        count = 0
-        my_each do |x|
-            count +=1 if yield x
-        end
-        count
-    end
-
-    def my_map(arr)
-        result = []
-        my_each(arr) do |x|
-            result.push(yield(x))
-        end
-        result
-    end
-
-
-
-
-
-
-
-end
\ No newline at end of file
+module Enumerable
+  def my_each(block)
+    result = []
+    return to_enum unless block_given?
+
+    (0..length).each do |i|
+      result.push(yield(block[i]))
+    end
+    result
+  end
+
+  def my_each_with_index(block)
+    result = []
+    return to_enum unless block_given?
+
+    (0..length).each do |i|
+      result.push(yield(block[i], i))
+    end
+    result
+  end
+
+  def my_select
+    result = []
+    return to_enum unless block_given?
+
+    my_each do |x|
+      output.push(x) if yield x
+    end
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each do |x|
+      result = false unless yield x
+    end
+    result
+  end
+  if attr.is_a? Class
+    my_each do |x|
+      return false unless x.is_a? attr
+    end
+    true
+  end
```

Comment:
It's not possible to have something like this.
You have to put it inside a method